src/app/main/component/events/components/events-list/events-list.component.ts

```diff
@@ -115,13 +116,23 @@ export class EventsListComponent implements OnInit, OnDestroy {
   }
 
   showSelectedEvents(): void {
-    this.bookmarkSelected = !this.bookmarkSelected;
-    if (this.bookmarkSelected) {
-      this.cleanEventList();
-      this.getUserFavoriteEvents();
+    if (!this.isLoggedIn) {
+      this.openAuthModalWindow('sign-in');
+      this.dialogRef
+        .afterClosed()
+        .pipe(take(1))
+        .subscribe((result) => {
+          this.isLoggedIn = !!result;
+        });
     } else {
-      this.cleanEventList();
-      this.getEvents();
+      this.bookmarkSelected = !this.bookmarkSelected;
+      if (this.bookmarkSelected) {
+        this.cleanEventList();
```

Comment:
try don't use nested if `bookmarkSelected` could be handled in a separate method to improve readability and separation of concerns:
```
private toggleEventList(): void {
  this.bookmarkSelected = !this.bookmarkSelected;
  this.cleanEventList();
 this.getEvents();
```
pay attention: duplicate Check on `(this.bookmarkSelected)`, once in `showSelectedEvents()` and again in `getEvents()`. The page variable is incremented in both , which could lead to unintended results if `getUserFavoriteEvents()` is directly called outside `getEvents()`.